### DISCUSSION TITLE
Cursor disappears occasionally - Manjaro Linux

### DISCUSSION THREAD

**Discussion by White-waluigi** [Genuitec/Copilot4Eclipse-community#5] [Help]
Whenever I activate this plugin, my text cursor (the blinking |) dissapears from the editor. I can click anywhere, but it doesn't come back. It only happens when I use this plugin. the only way to return the cursor is minimizing eclipse and opening it again. Any idea what might cause this or how to fix it? Or can I bind copilot to a key so it only shows up when I need it to?

**Comment by toddewilliams** [comment]
This is unusual and we haven't had it reported before. To help us figure this out, can you please post a full description of your environment?

OS version, window manager if Linux, Eclipse IDE and version, any UI or theming plugins, clean install or upgrade, etc.

**Comment by White-waluigi** [comment]
Pretty much vanilla Manjaro i3. I've tested it with a fresh installed os in a vm and it still happens. Runs X11, no plugins aside from copilot. Eclipse version is:

Version: 2024-06 (4.32.0)
Build id: 20240606-1231

Copilot Version is: 121.

It should happen very quickly when you start editing, the cursor will disappear sometimes when suggestions are made an not come back until you switch to another application.

**Comment by wayneparrott** [comment]
Thx for the info. We will give it a look and follow up asap. Copilot4Eclipse's editor integrations use std apis for programmatically updating cursor location. Perhaps the plugin may need to be a little more aggressive ui refreshing...
